tools/common-code/cmonitor_watcher.py

```diff
@@ -0,0 +1,196 @@
+        listOfFiles = os.listdir(dir)
+        allFiles = list()
+        for entry in listOfFiles:
+            fullpath = os.path.join(dir, entry)
+            if os.path.isdir(fullpath):
+                allFiles = allFiles + self.__get_list_of_files(fullpath)
+            else:
+                allFiles.append(fullpath)
+
+        return allFiles
+
+    def __process_task_files(self, dir):
+        """Process all the files for triggered-event within the watched dir.
+            Finds the process Ids and filter out the process name against the
+            provided white-list. If the process Id matches the whilte-listing
+            process from command-line , it store and return the file anlog with the process-name.
+            The process name later will be used to get the ip and port from the
+            command-line for the specific process.
+        Args:
+          dir: dir to be watched.
+
+        Returns:
+          The file along with the process name which will be used to launch cmonitor.
+
+        """
+        # time.sleep(20)
+        time.sleep(self.timeout)
+        logging.info(f"watcher process file sleep: {self.timeout}")
+        allFiles = self.__get_list_of_files(dir)
+        for file in allFiles:
+            if file.endswith("tasks"):
+                list = self.__get_pid_list(file)
+                if list:
+                    for pid in list:
+                        process_name = self.__get_process_name(pid)
+                        logging.info(f"processing task file: {file} with pid: {pid}, process name: {process_name}")
+                        match = self.__check_filter(process_name)
+                        if match is True:
+                            logging.info(f"Found match: {process_name}")
+                            self.myFileList = {file: process_name}
+                            return self.myFileList
+
+    def __check_filter(self, process_name):
+        """Check process name against the whilte-list.
+        Args:
+          process_name: process name to be matched against the whilte-list from command-line.
+
+        Returns:
+          True if process_name matches with the white-list.
+
+        """
+        for e in self.filter:
+            if process_name in e:
+                return True
+
+    def inotify_events(self, queue):
+        """Main thread function for notifying events.
+            Monitored events that match with the white-list provided will be stored in this queue.
+            The events from this queue will be processed by cMonitorLauncher threading function to
+            launch cMonitor with appropriate command input
+        Args:
+          queue: monitored events will be stored in this queue.
+
+        Returns:
+
+        """
+        logging.info(f"CgroupWatcher calling inotify_event")
+        i = inotify.adapters.Inotify()
+        i.add_watch(self.path)
+        try:
+            for event in i.event_gen():
+                if event is not None:
```

Comment:
add a comment like "if event is None, it means the Inotify() system has no more events that need to be processed... give control back to the caller"